.github/workflows/publish.yaml

```diff
@@ -70,21 +72,36 @@ jobs:
         if: ${{ always() && !inputs.snapshot }}
         uses: ./.github/actions/debug-output
 
+      # Determine repository to publish to
+      - name: Determine destination repository
+        id: repo
+        run: |
+          # Publish unicorn flavor to private repository
+          if [ "${{ matrix.flavor }}" = "unicorn" ]; then
+            if [ "${{ inputs.snapshot }}" = "true" ]; then
+              echo "Publishing unicorn flavor snapshot to ghcr.io/defenseunicorns/packages/private/uds/snapshots"
+              echo "repo=ghcr.io/defenseunicorns/packages/private/uds/snapshots" >> "$GITHUB_OUTPUT"
+            else
+              echo "Publishing unicorn flavor to ghcr.io/defenseunicorns/packages/private/uds"
+              echo "repo=ghcr.io/defenseunicorns/packages/private/uds" >> "$GITHUB_OUTPUT"
+            fi
+          # Publish snapshots to snapshot repository
+          elif [ "${{ inputs.snapshot }}" = "true" ]; then
+            echo "Publishing snapshot to ghcr.io/defenseunicorns/packages/uds/snapshots"
+            echo "repo=ghcr.io/defenseunicorns/packages/uds/snapshots" >> "$GITHUB_OUTPUT"
+          # Publish all other packages/bundles to uds package repository
+          else
+            echo "Publishing packages and bundles to ghcr.io/defenseunicorns/packages/uds"
+            echo "repo=ghcr.io/defenseunicorns/packages/uds" >> "$GITHUB_OUTPUT"
+          fi
```

Comment:
If its not too much of a pain you could build `repo` as a variable and end up with something like:
```
repo=ghcr.io/defenseunicorns/packages

if (unicorn) {
  repo += /private
} 

repo += /uds

if (snapshot) {
  repo += /snapshots
}
```

Reply:
I like that, will update and validate on my testing fork.

Reply:
nit: I would also consider if we should just move this logic to compute the TARGET_REPO into the publish task?  Overall the dumber our GH actions are the better I think but understand it would be a trade off passing a bunch of inputs around.

Reply:
Modified the repo construction var and updated the test pipeline links in the PR description. On moving the logic to the publish task I am leaning towards no at least for now:
- Publish should only ever be done in CI/actions so I'm not as concerned with that logic being actions specific
- With the refactor you suggested this logic is a lot simpler and very portable if we did need to move to GitLab or some other CI system